docs/how-to-migrate-from-smartlock.md

```diff
@@ -0,0 +1,791 @@
+  } else {
+    Log.w(TAG, "Smart Lock returned unknown credential type, ignoring");
+    startManualAuthencation();
+  }
+}
+```
+
+As can be seen, the code is structurally very similar. The properties on the
+Smart Lock credential type map to the OpenYOLO credential type as follows:
+
+|--------------------------------|------------------------------|--------------|
+| *Smart Lock Credential method* | *OpenYOLO Credential method* | *Notes*      |
+|--------------------------------|------------------------------|--------------|
+| getId()                        | getIdentifier()              |              |
+|--------------------------------|------------------------------|--------------|
+| getAccountType()               | getAuthenticationMethod()    | Both are     |
+|                                |     | URI-based, but OpenYOLO does not      |
+|                                |     | leave password auth as an implicit    |
+|                                |     | "other option"; all OpenYOLO          |
+|                                |     | credentials must have a defined       |
+|                                |     | authentication method, which makes    |
+|                                |     | processing the credentials easier.    |
+|------------------------|---------------------------|-------------------------|
+| getPassword()          | getPassword()             |                         |
+|------------------------|---------------------------|-------------------------|
+| getName()              | getDisplayName()          |                         |
+|------------------------|---------------------------|-------------------------|
+| getProfilePictureUri() | getDisplayPicture()       |                         |
+|------------------------|---------------------------|-------------------------|
+| getIdTokens()          | getIdToken()              |                         |
+|------------------------|---------------------------|-------------------------|
+| getGeneratedPassword() | N/A  | As OpenYOLO credentials are specific to the  |
+|                        |      | existing account case, generated passwords   |
+|                        |      | are not applicable.                          |
+|------------------------|------|----------------------------------------------|
+| getFamilyName()        | N/A  | Not defined in the OpenYOLO specification,   |
+|                        |      | as this assumes too much culture-specific    |
+|                        |      | structure to names.                          |
+|------------------------|------|--------------------------------------------- |
+| getGivenName()         | N/A  | Not defined in the OpenYOLO specification,   |
+|                        |      | for the same reasons as for `getFamilyName`. |
+|-----|---------------------------|--------------------------------------------|
+| N/A | getAdditionalProperties() | OpenYOLO offers extensibility for Hints    |
```

Comment:
Looks like a copy/paste error... should be `Credentials` rather than `Hints`

Reply:
Done.